tests/cpp17/generated_cpp17/monster_test_generated.h

```diff
@@ -1439,6 +1457,15 @@ struct Monster FLATBUFFERS_FINAL_CLASS : private flatbuffers::Table {
     return GetPointer<const void *>(VT_TEST);
   }
   template<typename T> const T *test_as() const;
+  AnyVariant test_variant() const {
```

Comment:
There are various reasons why `std::variant` is not the greatest match for unions (an is worse in usage than the `struct $(NAME)Union` we're already emitting which are frankly identical in purpose:

1) as we see here, the variant has its own "enum" mapping of values to types that is private, so here we need a switch to map from one to the other. In our own `Union` this whole switch reduces to assignment.
2) `std::variant` has this weird design that it wants only valid values, whereas our unions "none" is a very legit value. This causes you to have to add this `monostate` to the list of types, which frankly, looks like a hack to me. Testing for this stuff like `std::holds_alternative<std::monostate>()` seems really unelegant too (compared to a simple `Empty` method we could supply).

You could argue that it is good to support `variant` as it is, well, standard, but a `variant` of all these FlatBuffers types is not going to be compatible with anything, so I wonder in what context it is better than the `Union` we already generate?

Note, just commenting that it is my personal sense that this is not improving on what we already have, but if you have good reasons why you or someone else need this, do not feel blocked by me to merge this.

Reply:
@vglavnyy maybe has opinions.

Reply:
You are right. So I guess my point is that a similar union struct for the base API would be more ergonomic than using `variant`, minus the fact that it is not "standard".

I don't get your point about the `switch`.. with `variant`, you either have to do a sequence of `if (std::holds_alternative<T>)..` or use `std::visit` like you say, both of which are more verbose / clumsy / less direct than the switch above.

Pretty much all unions I have ever seen are fully unique types, and even a `switch` makes it easy to handle to members as one case body.

Again, if you feel someone would prefer this, it doesn't hurt as an alternative I suppose. But to me it seems strictly worse than what we have.